cmd/prom-migrator/main.go

```diff
@@ -0,0 +1,195 @@
+			if ok {
+				log.Error("msg", fmt.Errorf("running writer: %w", err).Error())
+				os.Exit(2)
+			}
+			break loop
+		}
+	}
+
+	log.Info("msg", "migration successfully carried out")
+	log.Info("msg", "exiting!")
+}
+
+func parseFlags(conf *config, args []string) {
+	flag.StringVar(&conf.name, "progress-job-name", migrationJobName, "Name for the current migration that is to be carried out. It corresponds to the value of the label 'job' set inside the progress-metric-name.")
+	flag.Int64Var(&conf.mint, "mint", 0, "Minimum timestamp (in seconds) for carrying out data migration.")
+	flag.Int64Var(&conf.maxt, "maxt", time.Now().Unix(), "Maximum timestamp (in seconds) for carrying out data migration. Setting this value less than zero will indicate all data from mint upto now. "+
+		"Setting mint and maxt less than zero will migrate all data available in the read storage.")
+	flag.StringVar(&conf.maxBlockSize, "max-block-size", "500MB", "Maximum size of in-memory block. More the size of the block, more will be the data migrated by any single block.")
+	flag.StringVar(&conf.readURL, "read-url", "", "URL address for the storage where the data is to be read from.")
+	flag.StringVar(&conf.writeURL, "write-url", "", "URL address for the storage where the data migration is to be written.")
+	flag.StringVar(&conf.progressMetricName, "progress-metric-name", progressMetricName, "Prometheus metric name for tracking the last maximum timestamp pushed to the remote-write storage. "+
+		"This is used to resume the migration process after a failure.")
+	flag.StringVar(&conf.progressMetricURL, "progress-metric-url", "", "Read URL of the write storage. This is used to fetch the progress-metric that represents the last pushed maximum timestamp.")
+	flag.BoolVar(&conf.progressEnabled, "progress-enabled", true, "This flag tells the migrator, whether or not to use the progress mechanism. It is helpful if you want to "+
+		"carry out migration with the same time-range. Without this, the migrator will resume the migration from the last time, where it as stopped.")
+	// Authentication.
+	// TODO: Auth/password via password_file and bearer_token via bearer_token_file.
+	flag.StringVar(&conf.readerAuth.Username, "read-auth-username", "", "Auth username for remote-read storage.")
+	flag.StringVar(&conf.readerAuth.Password, "read-auth-password", "", "Auth password for remote-read storage.")
+	flag.StringVar(&conf.readerAuth.BearerToken, "read-auth-bearer-token", "", "Auth username for remote-read storage. This should be mutually exclusive with username and password.")
+	flag.StringVar(&conf.writerAuth.Username, "write-auth-username", "", "Auth username for remote-write storage.")
+	flag.StringVar(&conf.writerAuth.Password, "write-auth-password", "", "Auth password for remote-write storage.")
+	flag.StringVar(&conf.writerAuth.BearerToken, "write-auth-bearer-token", "", "Auth username for remote-write storage. This should be mutually exclusive with username and password.")
```

Comment:
where are we validating that this is mutually exclusive?

Reply:
The validate() checks for mutual exclusion.

Reply:
```
if err := httpConfig.Validate();
```
this line is responsible for it.

Reply:
I have added tests for this in `main_test.go`. Please look for tests below the comment `// mutual exclusive tests` in that file.